components/utils/PopulationMultiSelect.tsx

```diff
@@ -0,0 +1,184 @@
+        if (
+          populationCode &&
+          !measurePopulations.map(p => p.value).includes(populationCode) &&
+          !EXCLUDED_MEASURES.includes(populationCode)
+        ) {
+          measurePopulations.push({
+            value: populationCode,
+            label: populationDisplay || populationCode,
+            disabled: false
+          });
+        }
+      });
+    });
+    return measurePopulations;
+  };
+
+  /**
+   * Sets desired populations state (for the selected patient) based on the measure
+   * populations selected from MultiSelect component and their relevant supset populations.
+   * @param value array of selected options from MultiSelect component
+   */
+  const updateDesiredPopulations = (value: string[]) => {
+    const newDesiredPopulations = value;
+    if (selectedPatient) {
+      // add intiial population since it is a superset of all other populations
+      if (value.length > 0 && !value.includes(Enums.PopulationType.IPP)) {
+        newDesiredPopulations.push(Enums.PopulationType.IPP);
+      }
+      // add denominator if numerator, numerator exclusion, or denominator exception are selected
+      if (
+        [Enums.PopulationType.NUMER, Enums.PopulationType.NUMEX, Enums.PopulationType.DENEXCEP].some(p =>
+          value.includes(p)
+        ) &&
+        !value.includes(Enums.PopulationType.DENOM)
+      ) {
+        newDesiredPopulations.push(Enums.PopulationType.DENOM);
+      }
+
+      // update the desired populations state
+      produce(currentPatients, async draftState => {
+        draftState[selectedPatient].desiredPopulations = newDesiredPopulations;
+      }).then(nextPatientState => {
+        setCurrentPatients(nextPatientState);
+      });
+      // update value that appears in MultiSelect component
+      setValue(newDesiredPopulations);
+    }
+  };
+
+  /**
+   * Uses the selected populations from the MultiSelect component to determine which
+   * populations can be selected, based on the population criteria.
+   * If a population cannot be selected in conjunction with a population that has already been selected,
+   * the population will become disabled in the dropdown.
+   *
+   * Based on population criteria for proportion measures:
+   * https://build.fhir.org/ig/HL7/cqf-measures/measure-conformance.html#population-criteria
+   *
+   * @param populations all measure populations available for the measure
+   * @returns array of filtered populations that can be selected
+   */
+  const updateMeasurePopulations = (populations: MultiSelectData[]): MultiSelectData[] => {
+    // disable denominator exclusion if denominator selected
+    if (value.includes(Enums.PopulationType.DENOM)) {
+      const denex = populations.find(e => e.value === Enums.PopulationType.DENEX);
+      if (denex) denex.disabled = true;
+    }
+
+    // disable all other populations (aside from IPP) if denominator exclusion selected
```

Comment:
When the user selects Denominator Exclusion, only the Initial Population shows up. I think in order to make sure that the Denominator Exclusion shows up, you have to disable all other populations (aside from IPP AND denominator exclusion).

![Screen Shot 2022-09-21 at 9 46 03 AM](https://user-images.githubusercontent.com/30158156/191570657-838404a3-1c7b-4531-ac32-3c4a27a1b429.png)